### PR TITLE
Add Github release support

### PR DESCRIPTION
**Pull request by clamoriniere1A**
Fixes: #25

This project is using [goreleaser](https://goreleaser.com/) and an additional script for releasing also the Helm chart.

For starting the delivery, you need to clone this repository, then:

```
zsh hack/release.sh <version> <remove-git>
```

a concreate example is: ```zsh hack/release.sh v1.0.1 upstream```

This script will:
- generates locally the helm chart with the requested version.
- updates the helm repo index file (```index.yaml```) file with the new release.
- Adds commits and tag the repository with all change and the requested version.
- Push changeset and tag to the remote git repository.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,13 +22,38 @@ install:
...
+            - "docs/helm-test-$TAG.tgz"
+          on:
+            tags: true
+            branches:
+              only:
+                - master
+                - /^v\d+(\.\d+)+$/
 after_success:
-  - bash <(curl -s https://codecov.io/bash)
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (sdminonne) [inline .travis.yml]: as usual the '\n' :)

File: hack/release.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+fi
+
+zsh $GIT_ROOT/hack/helm-release.sh $1
+
+# Update CHANGELOG.md file
+ssed -i.bak "5i## Release $1\n" CHANGELOG.md
+
+git commit -am "release $1"
+git tag -f $1
+git push -f --tags $GIT_REMOTE master
```
Comment (sdminonne) [inline hack/release.sh]: same

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - image: workflowcontroller/workflow-controller
+    goos: linux
+    goarch: amd64
+    goarm: ''
+    binary: workflow-controller
+    latest: true
+snapshot:
+  name_template: SNAPSHOT-{{ .Commit }}
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_checksums.txt'
```
Comment (sdminonne) [inline .goreleaser.yml]: same

Comment (clamoriniere1A) [reply .goreleaser.yml]: I will fix it with this settings in vscode ` "files.insertFinalNewline": true, ` :)
https://github.com/Microsoft/vscode/issues/12076

**Comment by clamoriniere1A** [comment]
@sdminonne @EricMountain-1A Finally I was able to run the e2e test in travis-ci.
In order to achieve it, I replaced the `kubeadm-dind-cluster` with `minikube`.

**Pull request merged**